// aylien-news-api/dist/model/Logicals.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AnyOfLogicalsParameter = _interopRequireDefault(require("./AnyOfLogicalsParameter"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Logicals model module.
 * @module model/Logicals
 * @version 5.1.1
 */
var Logicals = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Logicals</code>.
   * @alias module:model/Logicals
   */
  function Logicals() {
    _classCallCheck(this, Logicals);

    Logicals.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Logicals, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Logicals</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Logicals} obj Optional instance to populate.
     * @return {module:model/Logicals} The populated <code>Logicals</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Logicals();

        if (data.hasOwnProperty('$and')) {
          obj['$and'] = _ApiClient["default"].convertToType(data['$and'], [_AnyOfLogicalsParameter["default"]]);

          if ('$and' !== 'and') {
            Object.defineProperty(obj, 'and', {
              get: function get() {
                return obj['$and'];
              }
            });
          }
        }

        if (data.hasOwnProperty('$or')) {
          obj['$or'] = _ApiClient["default"].convertToType(data['$or'], [_AnyOfLogicalsParameter["default"]]);

          if ('$or' !== 'or') {
            Object.defineProperty(obj, 'or', {
              get: function get() {
                return obj['$or'];
              }
            });
          }
        }

        if (data.hasOwnProperty('$not')) {
          obj['$not'] = _ApiClient["default"].convertToType(data['$not'], [_AnyOfLogicalsParameter["default"]]);

          if ('$not' !== 'not') {
            Object.defineProperty(obj, 'not', {
              get: function get() {
                return obj['$not'];
              }
            });
          }
        }
      }

      return obj;
    }
  }]);

  return Logicals;
}();
/**
 * @member {Array.<module:model/AnyOfLogicalsParameter>} $and
 */


Logicals.prototype['$and'] = undefined;
/**
 * @member {Array.<module:model/AnyOfLogicalsParameter>} $or
 */

Logicals.prototype['$or'] = undefined;
/**
 * @member {Array.<module:model/AnyOfLogicalsParameter>} $not
 */

Logicals.prototype['$not'] = undefined;
var _default = Logicals;
exports["default"] = _default;
